### DISCUSSION TITLE
Define an environment variable in terms of other enviroment variables

### DISCUSSION THREAD

**Discussion by mrwilby** [nushell/nushell#10222] [Q&A] [answered]
In my zsh configuration, I have something like this:

`MAVEN_OPTS = '-Djavax.net.ssl.keyStore=${KEY_STORE_PATH}`

where `KEY_STORE_PATH` is another environment variable with a specific value.

Is this paradigm supported in nushell? I can workaround it by just defining my `$env.MAVEN_OPTS` variable and expanding it manually, but I'd prefer not to if this is somehow doable out-the-box.

**Comment by amtoine** [comment, marked as answer]
in Nushell you would do
```nushell
$env.MAVEN_OPTS = $"-Djavax.net.ssl.keyStore=($env.KEY_STORE_PATH)"
```
:yum: 

i.e.
- environment variables are set with the `$env.FOO = ...` syntax
- accessing an environment variable is then just `$env.FOO`
- here i use *string interpolation* to *insert* the first variable into the second, the general syntax is `$"($variable) ... (some | command)"`

**Reply by mrwilby** [reply]
Wonderful. I tried something similar but didn't spot the $" to evaluate. Much appreciate the quick help!

**Comment by amtoine** [comment]
> Wonderful. I tried something similar but didn't spot the $" to evaluate.
>
it's mentionned [there](https://www.nushell.sh/book/working_with_strings.html#string-interpolation) in the book :wink: 

> Much appreciate the quick help!
>
happy to help :blush:
